app/controllers/teachers/registrations_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class Teachers::RegistrationsController < Devise::RegistrationsController
+  before_action :configure_sign_up_params, only: [:create]
+  before_action :configure_account_update_params, only: [:update]
+
+  # GET /resource/sign_up
+  # def new
+  #   super
+  # end
+
+  # POST /resource
+  def create
+    @teacher = Teacher.new(sign_up_params)
+    if @teacher.save
+      redirect_to root_path
+      flash[:notice] = 'Account has been successfully created'
+    else
+      super
+    end
+  end
+
+  # GET /resource/edit
+  # def edit
+  #   super
+  # end
+
+  # PUT /resource
+  # def update
+  #   super
+  # end
+
+  # DELETE /resource
+  # def destroy
+  #   super
+  # end
+
+  # GET /resource/cancel
+  # Forces the session data which is usually expired after sign
+  # in to be expired now. This is useful if the user wants to
+  # cancel oauth signing in/up in the middle of the process,
+  # removing all OAuth session data.
+  # def cancel
+  #   super
+  # end
+
+  protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+  def configure_sign_up_params
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[first_name last_name])
+  end
+
+  # If you have extra params to permit, append them to the sanitizer.
+  def configure_account_update_params
+    devise_parameter_sanitizer.permit(:account_update, keys: %i[first_name last_name])
+  end
```

Comment:
Why is this code needed? 🤔

Reply:
nevermind, I figured out that `RegistrationsController` has an `edit` method which lets you edit your profile as a User. I didn't know that, thanks for teaching me this Marta! :)

Reply:
This is the same issue as the rest of sanitizer